sky/clouds/cloud.py

```diff
@@ -189,9 +189,27 @@ def check_credentials(self) -> Tuple[bool, Optional[str]]:
         """
         raise NotImplementedError
 
-    def get_cloud_user_identity(self) -> Optional[str]:
-        """Returns the identity of the user on this cloud.
+    def get_current_user_identity(self) -> Optional[str]:
+        """(Advanced) Returns currently active user identity of on this cloud.
 
+        The user "identity" is associated with each SkyPilot cluster they
+        creates. This is used in protecting cluster operations in a multi-
+        identity scenario, where the same user/device can switch between
+        different cloud identities.
+
+        The choice of what constitutes an identity is up to each cloud's
+        implementation. In general, to suffice for the above purposes,
+        ensure that different identities imply different resources are
+        displayed in the cloud's console.
+
```

Comment:
> Is it ok for a new identity to autostop a cluster?

I have thought about it. I think it should be fine to allow a new identity to autostop the cluster because:
1. The cluster is all launched by the user. I think the user should have the maximum permission to operate the cluster.
2. Autostop will work, as the cluster has the correct identity and can stop itself.

Reply:
Make sense to me.

Can we add (1) when is identity check performed
> We now check the identity only when the code modifies the cluster or has some cloud API operations. Including the following places:
> 
> 1. Before provisioning on an existing cluster;
> 2. Before stopping/terminating a cluster;
> 3. Before refreshing the status of a cluster.
> 
> Other operations that go through `ssh` will not be checked so that users can still interact with those clusters after switching the cloud account. (If the cluster is a spot instance or has autostop/setup, most operations will still fail, as the status refresh will be triggered before them.)

and (2) this comment about the design choice of "the user should have the maximum permission to operate the cluster." + "as long as it'll work, we allow it" here?